tests/ui/macros/macro-missing-fragment-deduplication.rs

```diff
@@ -2,12 +2,11 @@
 
 macro_rules! m {
     ($name) => {}; //~ ERROR missing fragment
-                   //~| ERROR missing fragment
 }
 
 fn main() {
-    m!();
-    m!();
-    m!();
-    m!();
+    m!(); //~ ERROR unexpected end
+    m!(); //~ ERROR unexpected end
+    m!(); //~ ERROR unexpected end
+    m!(); //~ ERROR unexpected end
```

Comment:
I don't know why these new "unexpected end" errors are showing up at the invocation rather than at `macro_rules!` or not at all. Any idea how to get rid of this?

Reply:
> I don't know why these new "unexpected end" errors are showing up at the invocation rather than at `macro_rules!` or not at all. Any idea how to get rid of this?

Well, `m` now expects a `tt` in the input due to recovery, but `m!()` do not pass a `tt`.

Reply:
That all makes sense, thanks for the explanation. I'm happy to change to either or just leave it as-is, what is your preferred solution here?

Reply:
Let's just use `tt` as the recovery for all erroneous fragment kinds.